### PR TITLE
fileio: do not try to read virtual file 3 times when max_size is small enough

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/test/test-fileio.c [hunk 1]
```diff
@@ -1066,6 +1066,7 @@ int main(int argc, char *argv[]) {
         test_read_nul_string();
         test_read_full_file_socket();
         test_read_full_file_offset_size();
+        test_read_virtual_file(0);
```
Comment (poettering) [inline src/test/test-fileio.c:1069]: maybe also add an invocation with == 1 here?

Comment (poettering) [reply src/test/test-fileio.c:1069]: and maybe even 2?

**Comment by poettering** [comment]
So, I think it would be simpler to just always check if the bytes we read already satisfy size_max, and return then. I made such a change in #19691, ptal

**Comment by yuwata** [comment]
Yeah, thank you.

**Pull request closed without merging**